modules/standard/online/online_controller.py

```diff
@@ -168,23 +168,73 @@ def online_discord_cmd(self, reply, args):
         blob = ""
         count = 0
 
+        # iterate channels
         for channel in [self.ORG_CHANNEL, self.PRIVATE_CHANNEL]:
+            # get characters
             online_list = self.get_online_characters(channel)
 
-            if len(online_list) > 0:
-                blob += "\n[%s]\n" % channel
-                current_main = ""
-                for row in online_list:
-                    if current_main != row.main:
-                        count += 1
-                        current_main = row.main
+            # if no characters, skip this channel
+            if len(online_list) == 0:
+                continue
+
+            # used to keep track of the current main
+            current_main = ""
+
+            # store maximum string lengths, these will be used to space the result table
+            max_lengths = [0, 0, 0, 0]
+
+            # hold values for rendering later
+            online_list_processed = []
+
+            # iterate online players in this channel
+            for row in online_list:
+                # increment count based on mains only
+                if current_main != row.main:
+                    count += 1
+                    current_main = row.main
+
+                # create array of 'processed' character information
+                processed = [
+                    row.name,
+                    "(%d/%d)" % (row.level or 0, row.ai_level or 0),
+                    row.faction,
+                    row.profession
+                ]
+
+                # append to other array for iteration later
+                online_list_processed.append(processed)
+
+                # iterate character information
+                for i in range(4):
```

Comment:
instead of `range(4)` this should be `range(len(processed))` which will make it easier to change if we ever add or remove columns